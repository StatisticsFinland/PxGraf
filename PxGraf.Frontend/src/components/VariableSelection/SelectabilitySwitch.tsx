import React from 'react';
import { useTranslation } from 'react-i18next';

import { FormControlLabel, FormControl, Switch } from '@mui/material';

interface ISelectabilitySwitchProps {
    selected: boolean,
    onChange: (newValue: boolean) => object
}

export const SelectabilitySwitch: React.FC<ISelectabilitySwitchProps> = ({ selected, onChange }) => {
    const { t } = useTranslation();

    return (
        <FormControl fullWidth>
            <FormControlLabel
                control={
                    <Switch
                        checked={selected ?? false}
                        onChange={(event) => onChange(event.target.checked)}
                    />
                }
                label={t("variableSelect.selectable")}
            />
        </FormControl>
    )
}

export default SelectabilitySwitch;