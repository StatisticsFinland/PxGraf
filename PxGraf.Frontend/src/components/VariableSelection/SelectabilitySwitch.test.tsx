import { render } from "@testing-library/react";
import React from "react";
import SelectabilitySwitch from "./SelectabilitySwitch";

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => { }),
            },
        };
    },
}));

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<SelectabilitySwitch
            selected={true}
            onChange={jest.fn()}
        ></SelectabilitySwitch>);
        expect(asFragment()).toMatchSnapshot();
    });
});