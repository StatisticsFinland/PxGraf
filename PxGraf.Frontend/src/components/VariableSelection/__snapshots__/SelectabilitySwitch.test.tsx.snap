// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        variableSelect.selectable
      </span>
    </label>
  </div>
</DocumentFragment>
`;
